modules/util/url.go

```diff
@@ -25,6 +25,9 @@ func PathEscapeSegments(path string) string {
 
 // URLJoin joins url components, like path.Join, but preserving contents
 func URLJoin(base string, elems ...string) string {
+	// We do need to escape special chars here or else they can be silently discarded
+	// in the ResolveReference call below
+	base = PathEscapeSegments(base)
```

Comment:
If `base` requires escaping of its components, maybe `elems` need that too?

Reply:
Good catch, fixed